test/functional/test_framework/address.py

```diff
@@ -100,3 +128,10 @@ def check_script(script):
     if (type(script) is bytes or type(script) is CScript):
         return script
     assert False
+
+
+class TestFrameworkScript(unittest.TestCase):
+    def test_base58encodedecode(self):
+        self.assertEqual(base58_to_byte(byte_to_base58(b'\x1f\x8e\xa1p*{\xd4\x94\x1b\xca\tA\xb8R\xc4\xbb\xfe\xdb.\x05', 111), 111), b'\x1f\x8e\xa1p*{\xd4\x94\x1b\xca\tA\xb8R\xc4\xbb\xfe\xdb.\x05')
+        self.assertEqual(base58_to_byte(byte_to_base58(b':\x0b\x05\xf4\xd7\xf6l;\xa7\x00\x9fE50)l\x84\\\xc9\xcf', 111), 111), b':\x0b\x05\xf4\xd7\xf6l;\xa7\x00\x9fE50)l\x84\\\xc9\xcf')
+        self.assertEqual(base58_to_byte(byte_to_base58(b'A\xc1\xea\xf1\x11\x80%Y\xba\xd6\x1b`\xd6+\x1f\x89|c\x92\x8a', 111), 111), b'A\xc1\xea\xf1\x11\x80%Y\xba\xd6\x1b`\xd6+\x1f\x89|c\x92\x8a')
```

Comment:
In commit "tests: implement base58_decode" (31cfd81e1f374a722ab0190cd89814f22b8f91f8)

Could avoid needing to repeat the byte strings (they are hard to compare visually) with a helper lambda: 

```python
check = lambda data, ver: self.assertEqual(base58_to_byte(byte_to_base58(data, ver)), (data, ver))
check(b'\x1f\x8e\xa1p*{\xd4\x94\x1b\xca\tA\xb8R\xc4\xbb\xfe\xdb.\x05', 111)
check(b':\x0b\x05\xf4\xd7\xf6l;\xa7\x00\x9fE50)l\x84\\\xc9\xcf', 111)
check(b'A\xc1\xea\xf1\x11\x80%Y\xba\xd6\x1b`\xd6+\x1f\x89|c\x92\x8a', 111)
```

Also it might be good to add a check with 0 version number and some leading \0 bytes to make sure roundtrips with padding are working